### ISSUE TITLE
[Bug Report][3.0.0] V-App-Bar Collapse On Reload Page

### ISSUE THREAD

**Issue by BlackYuzia** [vuetifyjs/vuetify#15202]
### Environment
**Vuetify Version:** 3.0.0-beta.3
**Vue Version:** 3.2.36
**Browsers:** Chrome 102.0.0.0
**OS:** Windows 10

### Steps to reproduce
0. You should run server (npm install && npm run dev OR npm install && npm run build && npm run start)
1. Open `/` page
2. Reload page
3. Look at app bar
4. Bar collapse for a few seconds (in run dev mode this collapse a bit longer)

### Expected Behavior
App bar didn't collapse or something, duo page reload

### Actual Behavior
Bar collapse on every page reload. After a few seconds, bar return to valid position. If you disable javascript, app bar didn't return to required position.

### Reproduction Link
<a href="https://gitlab.com/BlackYuzia/nuxt-vuetify" target="_blank">https://gitlab.com/BlackYuzia/nuxt-vuetify</a>

### Screenshots

Collapse example:
![image](https://user-images.githubusercontent.com/25010528/171830429-cbc7afd1-0942-4479-b95f-fb3059b60396.png)

Un Collapsed after a few seconds:
![image](https://user-images.githubusercontent.com/25010528/171830435-6aafcd7e-7516-4877-8c49-180b82138c0f.png)

**Comment by besnikh** [comment]
I have the same issue working on Nuxt 3 and Vuetify 3.0.0-beta.3 
Is there any workaround ? 
I tried to use the <Suspense> but it's not working in Nuxt, I tried in layouts and App.vue as well

Cheers

**Comment by drftls** [comment]
Same issue here. Nuxt 3 and Vuetify 3.0.0-beta.5. Similar issue if I tried replacing it with a v-toolbar instead (except v-toolbar starts bigger then contracts). Even without a drawer, it still starts each reload in the small state before expanding to the full width of the screen once the page is fully loaded.

Using a card as a workaround for now.

**Comment by MrPalixir** [comment]
Same issue too with v-navigation-drawer.

**Comment by richardordinario** [comment]
this is the workaround I make

<script setup>
const loading = ref(true)
onMounted(() => {
  loading.value = false
})
</script>

<template>
 <v-app v-if="!loading"></app>
</template>

**Comment by richardordinario** [comment]
`
This is the workaround I make using layouts
"
<template>
 <v-app v-if="!loading"></app>
</template>
"
<script setup>
const loading = ref(true)
onMounted(() => {
  loading.value = false
})
</script>
`

**Comment by ChristopheCVB** [comment]
```html
<v-app-bar style="width: 100%">
```
Is what I use as a workaround

**Comment by cheetamaru** [comment]
Workaround for height collapsing:

```vue
<template>
    <div>
        <v-app>
            <v-app-bar ... style="width: 100%">
            ...
            </v-app-bar>
            <v-fade-transition leave-absolute>
                <div v-if="isExtraSpaceAdded" style="height: 48px" />
            </v-fade-transition>
            ...
        </v-app>
    </div>
</template>

<script setup>
const isExtraSpaceAdded = ref(true)

onMounted(() => {
  isExtraSpaceAdded.value = false
})
</script>
```

`heigth: 48px` is a heigth value of the `v-app-bar`.

Hope this bug will be fixed soon.

**Comment by pavloniym** [comment]
Still waiting ...
@cheetamaru workaround is working, but it is a hack ...

**Comment by johnleider** [comment]
Added `100%` width to `v-toolbar in [c76dc1a](vscode-file://vscode-app/c:/Users/johnj/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html). Still waiting on suspense for SSR.

**Comment by johnleider** [comment]
This bug is resolved as of the latest version.

If you have any questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com).

**Comment by BlackYuzia** [comment]
again and again and again? :eyes:

**Comment by KaelWD** [comment]
This is still blocked by https://github.com/vuejs/core/pull/6095

**Comment by mostafaznv** [comment]
I have the same issue working on nuxt@3.6.3 and vuetify@3.3.8

**Comment by richardordinario** [comment]
> I have the same issue working on nuxt@3.6.3 and vuetify@3.3.8 Is there any workaround?

try using <client></client>

**Comment by KaelWD** [comment]
If you're wrapping the whole page in `<client>` you might as well just disable ssr entirely.

**Comment by mubaidr** [comment]
For you can simply set following styles on `VMain` to overcome this issue: 

```html
<VApp
  <VAppBar>
    ... your content here
  </VAppBar>
  <VMain
      style="
        --v-layout-left: 0px;
        --v-layout-right: 0px;
        --v-layout-top: 64px;
        --v-layout-bottom: 44px;
      "
    >
    ... your content here
  </VMain>
</VApp>
```

**Comment by johnleider** [comment]
This should be better on newer versions, but we're still waiting on upstream.

**Comment by kingyue737** [comment]
https://github.com/vuejs/core/pull/7290 is released in vue 3.4 rc.1
Maybe it can be fixed now?

**Comment by devonik** [comment]
Mh i tried it with vue 3.4.0 rc3 but still weird rendering https://stackblitz.com/edit/github-wudebh-ppjcxg?file=plugins%2Fvuetify.ts

**Comment by kingyue737** [comment]
@devonik https://github.com/vuetifyjs/vuetify/pull/15229 has not been merged

**Comment by devonik** [comment]
ui thanks there are so many issues regarding this. Must be a good feeling to able to finish them all

**Comment by devonik** [comment]
Any news for this ?

**Comment by johnleider** [comment]
Working on it. Since we use TSX, Vue 3.4 requires changes on our end.

**Comment by angelorc** [comment]
We have the same bug

Link: https://bitsong.studio
Repo: https://github.com/bitsongofficial/studio

https://github.com/vuetifyjs/vuetify/assets/1754055/33e7a6a2-59ec-489b-af17-e10e77a22644

**Comment by Luferov** [comment]
This bug is still exist

![image](https://github.com/user-attachments/assets/80e66c2c-971b-433d-bcf8-86b8df0e4a03)

**Comment by johnleider** [comment]
Please create a new issue. https://issues.vuetifyjs.com/

**Comment by Luferov** [comment]
> Please create a new issue. https://issues.vuetifyjs.com/

Can't do this

![image](https://github.com/user-attachments/assets/48efd371-44cf-43d4-a467-6bb49d63e7e4)

**Comment by pavloniym** [comment]
On 3.7 - this bug exists

**Comment by KaelWD** [comment]
Suspense is still broken, we had to revert #15229.

**Comment by imduchy** [comment]
I'm facing the same (similar?) issue with Nuxt v4 and VAppBar on a pre-rendered (`"/": { prerender: true },`) page.

Reload...
<img width="3024" height="920" alt="Image" src="https://github.com/user-attachments/assets/68065758-2552-4ce9-ab43-ee72c35e77cf" />

After a couple of milliseconds...
<img width="3024" height="920" alt="Image" src="https://github.com/user-attachments/assets/72255fcd-3860-423d-afd3-32aa1673defb" />

**Comment by Axel1987** [comment]
This entire layout shift bug comes from the fact that Vuetify initializes the layout offsets with 0 instead of null.

If the initial value of --v-layout-top were null (or simply not rendered at all), the browser wouldn’t apply any offset on first paint, and the later recalculation during hydration wouldn’t cause a visible jump.

Right now the sequence is:

SSR renders with --v-layout-top: 0px;

On client hydration Vuetify updates it to 64px;
→ which immediately causes a layout shift.

Replacing <v-app-bar> with <v-toolbar> fixes the issue entirely, because v-toolbar doesn’t participate in the layout engine or set --v-layout-* variables.

In short — a 7-character fix (app-bar → toolbar) avoids a bug that wouldn’t exist if Vuetify initialized layout offsets as null instead of 0.

Please tag the core developers so this can be reviewed and addressed properly.

**Comment by J-Sek** [comment]
The problem does not occure when using `ssr: true` as described in [docs](https://vuetifyjs.com/en/getting-started/installation/#ssr-caveats).

If the problem still occurs, remix [this example](https://github.com/J-Sek/vuetify-vite-ssr) so we have a reproduction.
